lib/collection/src/shards/transfer/snapshot.rs

```diff
@@ -188,10 +189,17 @@ pub(super) async fn transfer_snapshot(
         "Local shard must be a queue proxy"
     );
 
+    let snapshot_storage_manager = Collection::create_snapshot_storage_manager(snapshots_path);
```

Comment:
The fact that we are using `create_snapshot_storage_manager` from shards suggesrts that it should not be part of `Collection`.
Ideally shards should not know anything about Collection existance, otherwise it is circular dependency.

Reply:
Do you think that lib/common/common would be a good location?

Reply:
I would do it as a `::new` with config and path as arguments. Since we do have access to both in all cases

Reply:
I think we don't have access to StorageConfig (which will hold the info for S3/LocaFS) from the shard holder, it does have SharedStorageConfig, so we could update that structure I think :shrug:

Reply:
`transferring_shard` have access to `shared_storage_config`. We can have a method to create SnapshotManager in both Collection and ShardReplicaSet

Reply:
Yes but SharedStorageConfig https://github.com/qdrant/qdrant/blob/d0baa5ef4a4bc2c597b28d0e1f1011262be8ed4d/lib/collection/src/operations/shared_storage_config.rs#L17 does not have the needed info for handling snapshot transfer

Reply:
Good point. I would add this info into `SharedStorageConfig` then